nautobot/dcim/api/serializers.py

```diff
@@ -1043,13 +1105,90 @@ class Meta:
         fields = "__all__"
 
 
-class ControllerSerializer(NautobotModelSerializer, TaggedModelSerializerMixin):
+class ControllerSerializer(TaggedModelSerializerMixin, NautobotModelSerializer):
     class Meta:
         model = Controller
         fields = "__all__"
 
 
-class ControllerManagedDeviceGroupSerializer(NautobotModelSerializer, TaggedModelSerializerMixin):
+class ControllerManagedDeviceGroupSerializer(TaggedModelSerializerMixin, NautobotModelSerializer):
     class Meta:
         model = ControllerManagedDeviceGroup
         fields = "__all__"
+
+
+#
+# Modules
+#
+
+
+class ModuleBaySerializer(TaggedModelSerializerMixin, NautobotModelSerializer):
+    class Meta:
+        model = ModuleBay
+        fields = "__all__"
+        validators = []
+
+    def validate(self, data):
+        """Validate device and module field constraints for module bay."""
+        if data.get("parent_device") and data.get("parent_module"):
+            raise serializers.ValidationError("Only one of parent_device or parent_module must be set")
+        if data.get("parent_device"):
+            validator = UniqueTogetherValidator(
+                queryset=self.Meta.model.objects.all(), fields=("parent_device", "position")
+            )
+            validator(data, self)
+        if data.get("parent_module"):
+            validator = UniqueTogetherValidator(
+                queryset=self.Meta.model.objects.all(), fields=("parent_module", "position")
+            )
+            validator(data, self)
+        return super().validate(data)
```

Comment:
Just checking - `super().validate()` covers the case where *neither* `parent_device` nor `parent_module` are provided, right?

Reply:
That's correct. The model validation actually covers all of these but drf recommends doing the validation on the serializer in case you decide to stop using `ModelSerializer` 🤷